sdk/core/core-auth/src/tokenCredential.ts

```diff
@@ -59,6 +59,32 @@ export interface GetTokenOptions {
    * Allows specifying a tenantId. Useful to handle challenges that provide tenant Id hints.
    */
   tenantId?: string;
+
+  /**
+   * Options for Proof of Possession token requests
+   */
+  proofOfPossessionOptions?: {
+    /**
+     * Indicates whether to enable Proof of Possession (PoP) for the requested token.
+     */
+    isEnabled?: boolean;
+    /**
+     * The nonce value required for PoP token requests. 
+     * This is typically retrieved from the WWW-Authenticate header of a 401 challenge response.
+     * This is used in combination with {@link resourceRequestUri} and {@link resourceRequestMethod} to generate the PoP token.
+     */
+    nonce?: string
```

Comment:
Which fields are required here?  Are they all optional?

Reply:
Yes they are optional

Reply:
Only needed for PopToken support

Reply:
What would be the minimal required fields for PopToken support then?

Reply:
Yeah I think the pop options itself should be optional but the required options to use pop should not be. That way we can get rid of the `isEnabled` flag and just see if the required options are there or not

Reply:
> Yeah I think the pop options itself should be optional but the required options to use pop should not be. That way we can get rid of the `isEnabled` flag and just see if the required options are there or not

Updated design acc to this